src/lib/Microsoft.Health.Extensions.Fhir.R4/FhirClientFactory.cs

```diff
@@ -64,18 +65,29 @@ public FhirClient Create()
 
         private static FhirClient CreateClient(TokenCredential tokenCredential, ITelemetryLogger logger)
         {
+            EnsureArg.IsNotNull(tokenCredential, nameof(tokenCredential));
+
             var url = Environment.GetEnvironmentVariable("FhirService:Url");
             EnsureArg.IsNotNullOrEmpty(url, nameof(url));
             var uri = new Uri(url);
 
-            EnsureArg.IsNotNull(tokenCredential, nameof(tokenCredential));
-
             var fhirClientSettings = new FhirClientSettings
             {
                 PreferredFormat = ResourceFormat.Json,
             };
 
-            var client = new FhirClient(url, fhirClientSettings, new BearerTokenAuthorizationMessageHandler(uri, tokenCredential, logger));
+            FhirClient client;
+            try
+            {
+                client = new FhirClient(url, fhirClientSettings, new BearerTokenAuthorizationMessageHandler(uri, tokenCredential, logger));
+            }
+            catch (Exception ex)
+            {
+                client = null;
+                FhirServiceExceptionProcessor.ProcessException(ex, logger);
+            }
+
+            FhirServiceValidator.ValidateFhirService(client, logger);

```

Comment:
Should we check against the returned bool if it succeeded?

Reply:
If the above fails and client = null, can we avoid calling the ValidateFhirService()

Reply:
In this scenario, I don't think checking the returned bool is needed for any reason.